lamvery/actions/api.py

```diff
@@ -0,0 +1,262 @@
+from pygments.formatters import TerminalFormatter
+
+DEFAULT_MAPPING_TEMPLATE = '''
+##  See http://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-mapping-template-reference.html
+##  This template will pass through all parameters including path, querystring, header, stage variables, and context through to the integration endpoint via the body/payload
+#set($allParams = $input.params())
+{
+"body-json" : "$input.json('$')",
+"params" : {
+#foreach($type in $allParams.keySet())
+    #set($params = $allParams.get($type))
+"$type" : {
+    #foreach($paramName in $params.keySet())
+    "$paramName" : "$util.escapeJavaScript($params.get($paramName))"
+        #if($foreach.hasNext),#end
+    #end
+}
+    #if($foreach.hasNext),#end
+#end
+},
+"stage-variables" : {
+#foreach($key in $stageVariables.keySet())
+"$key" : "$util.escapeJavaScript($stageVariables.get($key))"
+    #if($foreach.hasNext),#end
+#end
+},
+"context" : {
+    "account-id" : "$context.identity.accountId",
+    "api-id" : "$context.apiId",
+    "api-key" : "$context.identity.apiKey",
+    "authorizer-principal-id" : "$context.authorizer.principalId",
+    "caller" : "$context.identity.caller",
+    "cognito-authentication-provider" : "$context.identity.cognitoAuthenticationProvider",
+    "cognito-authentication-type" : "$context.identity.cognitoAuthenticationType",
+    "cognito-identity-id" : "$context.identity.cognitoIdentityId",
+    "cognito-identity-pool-id" : "$context.identity.cognitoIdentityPoolId",
+    "http-method" : "$context.httpMethod",
+    "stage" : "$context.stage",
+    "source-ip" : "$context.identity.sourceIp",
+    "user" : "$context.identity.user",
+    "user-agent" : "$context.identity.userAgent",
+    "user-arn" : "$context.identity.userArn",
+    "request-id" : "$context.requestId",
+    "resource-id" : "$context.resourceId",
+    "resource-path" : "$context.resourcePath"
+    }
+}
+'''
+
+
+class ApiAction(BaseAction):
+
+    def __init__(self, args):
+        super(ApiAction, self).__init__(args)
+        self._remove = args.remove
+        self._write_id = args.write_id
+        self._stage = args.stage
+
+    def get_stage_name(self):
+        if self._stage is not None:
+            return self._stage
+        else:
+            return self._config.get_api_stage()
+
+    def get_cors(self):
+        cors = self._config.get_api_cors()
+        if cors is not None:
+            cors['origin'] = "'{}'".format(cors['origin'])
+            cors['methods'] = "'{}'".format(','.join(cors['methods']))
+            cors['headers'] = "'{}'".format(','.join(cors['headers']))
+        return cors
+
+    def action(self):
+        client = self.get_apigateway_client()
+        api_id = self._config.get_api_id()
+        stage = self.get_stage_name()
+        cors = self.get_cors()
+        api_conf = self._integrate_aws(
+            self._config.get_api_configuration(), stage, cors)
+        lambda_client = self.get_lambda_client()
```

Comment:
__[flake8]__
* [F841] local variable 'lambda_client' is assigned to but never used